### PR TITLE
Add parameters fetch_remote_resources, http_proxy_server and http_proxy_port

### PR DESCRIPTION
**Pull request by treydock**
These changes are to support https://github.com/theforeman/foreman_scap_client/pull/21

### CODE REVIEW

**Comment by alexjfisher** [comment]
Oops.  Sorry I had a similar change in progress, but never got round to opening a PR. https://github.com/alexjfisher/puppet-foreman_scap_client/commit/43821f05e11e19b418701f57b4f627c299898caa

**Comment by alexjfisher** [comment]
I've opened https://github.com/theforeman/puppet-foreman_scap_client/pull/45 anyway.  It's pretty much a duplicate.  I think we've defaulted `fetch_remote_resources` differently.

**Comment by treydock** [comment]
@alexjfisher Be handy if your PR is merged to also include change to write output to logger.  The message when remote sources are fetched is written to STDERR and will be sent as an email to root mail alias if that is setup.

File: manifests/init.pp [hunk 1]
```diff
@@ -48,14 +48,17 @@
...
-  $ensure               = 'present',
-  $ca_file              = $::foreman_scap_client::params::ca_file,
-  $host_certificate     = $::foreman_scap_client::params::host_certificate,
-  $host_private_key     = $::foreman_scap_client::params::host_private_key,
-  $foreman_repo_rel     = undef,
-  $foreman_repo_key     = 'https://yum.theforeman.org/RPM-GPG-KEY-foreman',
-  $foreman_repo_src     = undef,
-  $foreman_repo_gpg_chk = false,
+  $ensure                 = 'present',
+  $fetch_remote_resources = true,
```
Comment (xprazak2) [inline manifests/init.pp]: Could `false` be a default? I would like to keep the old behaviour as default.

File: templates/config.yaml.erb [hunk 2]
```diff
@@ -5,6 +5,21 @@
 :server: '<%= @server %>'
 :port: <%= @port %>
 
+# Should --fetch-remote-resources be added to `oscap xccdf eval` command
+:fetch_remote_resources: <%= @fetch_remote_resources %>
+
+# HTTP proxy server for downloading remote resources
+<%- if @http_proxy_server -%>
```
Comment (xprazak2) [inline templates/config.yaml.erb]: I think you can omit the conditions here and just have:
```
:http_proxy_server: <%= @http_proxy_server %>
```

Same for the `:http_proxy_port` below.

**Comment by xprazak2** [comment]
Works nicely, will need rebase.

Adding the new params into readme could be helpful.

**Comment by treydock** [comment]
Rebased and updates made based on comments.

**Comment by xprazak2** [comment]
Merging, thanks!

**Pull request merged**